patch-emscripten.sh

```diff
@@ -0,0 +1,7 @@
+#/usr/bin/env bash
+if test -f $1/manifold.js; then
+  # I still have no idea why this is needed...
+  sed -i 's/new Worker/new (ENVIRONMENT_IS_NODE ? global.Worker : Worker)/g' $1/manifold.js
```

Comment:
I have no idea why `global.Worker != Worker`, considering there are no `let Worker` or `var Worker` anywhere...

Also, `Worker` is something like this:
```
class extends EventTarget {
    constructor(url, options2) {
      super();
      this._vw_workerTarget = new EventTarget();
      this._vw_insideListeners = /* @__PURE__ */ new Map();
      this._vw_outsideListeners = /* @__PURE__ */ new Map();
      this._vw_messageQueue = [];
      this.onmessage = null;
      this.onmessageerror = null;
      this.onerror = null;
      const context = {
        onmessage: null,
        name: options2 == null ? void 0 : options2.name,
        close: () => this.terminate(),
        dispatchEvent: (event) => {
          return this._vw_workerTarget.dispatchEvent(event);
        },
```

while `global.Worker` is this:
```
class Worker extends EventEmitter {
  constructor(filename, options = kEmptyObject) {
    super();
    debug(`[${threadId}] create new worker`, filename, options);
    if (options.execArgv)
      validateArray(options.execArgv, 'options.execArgv');

    let argv;
    if (options.argv) {
      validateArray(options.argv, 'options.argv');
      argv = ArrayPrototypeMap(options.argv, String);
    }

    let url, doEval;
    if (options.eval) {
      if (typeof filename !== 'string') {
        throw new ERR_INVALID_ARG_VALUE(
          'options.eval',
          options.eval,
          'must be false when \'filename\' is not a string',
        );
      }
      url = null;
      doEval = 'classic';
```

`global.Worker` is the correct target in this case.

Reply:
it seems that nodejs dynamic import will define `Worker` for the module being imported, but I am still not sure why the Worker defined here is the wrong one.

Reply:
From your other comment it sounds like you think this might be a `vitest` issue? Have we opened anything on their forums?

Reply:
I think it is related to `vitest`, but I cannot make a minimal example yet and don't quite understand why it causes an issue, so I have not yet open anything there.